tests/bsim/bluetooth/mesh/src/test_persistence.c

```diff
@@ -455,12 +455,6 @@ static void provisioner_setup(void)
 		FAIL("Failed to add test_netkey (err: %d, status: %d)", err, status);
 	}
 
-	err = bt_mesh_cfg_cli_net_transmit_set(test_netkey_idx, TEST_PROV_ADDR,
```

Comment:
Cool that this random delay helped to avoid this workarounds!

Reply:
Unfortunately not all. This is still required: https://github.com/zephyrproject-rtos/zephyr/blob/main/tests/bsim/bluetooth/mesh/src/test_persistence.c#L558-L562

After removing device looses transport acks (with legacy transport) during virtual addresses configuration. It is possible to fix with magic delays, but the current fix looks less magic.
Probably it will be possible to remove after legacy transport is gone.